### PR TITLE
Test vectors and Server Secret updates

### PR DESCRIPTION
**Pull request by wtoorop**
This is a PR for the combined issues #2 and #8 .

### CODE REVIEW

File: draft-sury-toorop-dnsop-server-cookies.md [hunk 1]
```diff
@@ -245,6 +257,60 @@ Notice that Client-IP is used for hash generation even though it's not included
...
+
+> After stage 1 completed, all the servers in the anycast set have learned the
+  new Server Secret, and can verify Server Cookies constructed with it, but keep
+  generator Server Cookies with the old Server Secret.
+
+Stage 2
+: This stage starts either by explicit signalling from the operator, or after
+  a configurable period after which it is reasonable to assume that the Server
+  Cookie is present on all members from the anycast set.  A default value for
+  such a configurable time could be 5 minutes.
```
Comment (oerdnj) [inline draft-sury-toorop-dnsop-server-cookies.md]: I think the "configurable period" should be dropped and only explicit signalling from the operator method should be kept here.

File: draft-sury-toorop-dnsop-server-cookies.md [hunk 2]
```diff
@@ -245,6 +257,60 @@ Notice that Client-IP is used for hash generation even though it's not included
...
+> When entering Stage 2, servers start generating Server Cookies with the new
+  Server Secret. The previous Server Secret is not yet removed/forgotten about.
+
+> Server Cookies constructed with the both the new Server Secret and with
+  the previous Server Secret are considered valid when verifying.
+
+Stage 3
+: This stage starts either by explicit signalling from the operator, or after
+  a configurable period after which it is reasonable to assume that most
+  clients have learned the new Server Secret.
```
Comment (oerdnj) [inline draft-sury-toorop-dnsop-server-cookies.md:303]: Same here.  The signalling should be explicit.  The anycast deployments are very complex and it's better not to mix provisioning guideliness with the implementation guideliness.

File: draft-sury-toorop-dnsop-server-cookies.md [hunk 3]
```diff
@@ -245,6 +257,60 @@ Notice that Client-IP is used for hash generation even though it's not included
...
+
+> We RECOMMEND this period to be the longest TTL in the zones served by the
+  server plus half an hour, but it SHOULD be at least longer than the period
+  clients are allowed to use the same Server Cookie, which SHOULD be half an
+  hour, see (#timestampField).
```
Comment (oerdnj) [inline draft-sury-toorop-dnsop-server-cookies.md]: However, this recommendation should be kept.

**Review by oerdnj — CHANGES_REQUESTED** [review]

**Comment by oerdnj** [comment]
The rest looks good to me.

File: draft-sury-toorop-dnsop-server-cookies.md (hunk 3 above)
Comment (wtoorop) [reply draft-sury-toorop-dnsop-server-cookies.md]: addressed in commit @9cf8dbc

File: draft-sury-toorop-dnsop-server-cookies.md (hunk 1 above)
Comment (wtoorop) [reply draft-sury-toorop-dnsop-server-cookies.md]: addressed in commit @9cf8dbc

File: draft-sury-toorop-dnsop-server-cookies.md (hunk 2 above)
Comment (wtoorop) [reply draft-sury-toorop-dnsop-server-cookies.md:303]: addressed in commit @9cf8dbc

**Pull request merged**